geps/gep-1897.md

```diff
@@ -125,7 +125,231 @@ This GEP is the outcome of the TLS use cases #4 and #5 in
+)
+```
+
+## How a client behaves
+
+This table describes the effect that a TLSConnectionPolicy has on a Route.  There are only two cases where the
+TLSConnectionPolicy will signal a Route to connect to a backend using TLS, an HTTPRoute with a backend that is targeted
+by a TLSConnectionPolicy, either with or without listener TLS configured.  (There are a few other cases where it may be
+possible, but is purposely marked “not supported” due to a desire for less confusion on the assigned purpose of each of
+the protocol-affiliated types of Routes.)
+
+| Route Type | Gateway Config             | Backend is targeted by a TLSConnectionPolicy? | Connect to backend  with TLS? |
+|------------|----------------------------|-----------------------------------------------|-------------------------------|
+| HTTPRoute  | Listener tls               | Yes                                           | **Yes**                       |
+| HTTPRoute  | No listener tls            | Yes                                           | **Yes**                       |
+| HTTPRoute  | Listener tls               | No                                            | No                            |
+| HTTPRoute  | No listener tls            | No                                            | No                            |
+| TLSRoute   | Listener Mode: Passthrough | Yes                                           | No                            |
+| TLSRoute   | Listener Mode: Terminate   | Yes                                           | Not supported                 |
+| TLSRoute   | Listener Mode: Passthrough | No                                            | No                            |
+| TLSRoute   | Listener Mode: Terminate   | No                                            | No                            |
+| TCPRoute   | Listener TLS               | Yes                                           | Not supported                 |
+| TCPRoute   | No listener TLS            | Yes                                           | Not supported                 |
+| TCPRoute   | Listener TLS               | No                                            | No                            |
+| TCPRoute   | No listener TLS            | No                                            | No                            |
+| UDPRoute   | Listener TLS               | N/A                                           | No                            |
+| UDPRoute   | No listener TLS            | N/A                                           | No                            |
+| UDPRoute   | Listener TLS               | N/A                                           | No                            |
+| UDPRoute   | No listener TLS            | N/A                                           | No                            |
+| GRPCRoute  | N/A                        | N/A                                           | No                            |
```

Comment:
should probably be the same as HTTP?

Reply:
I'm not proposing any other changes in behavior except for HTTPRoute, so I'm not in a hurry to make GRPCRoute have the same behavior as HTTPRoute and be changed as well.

Reply:
That feels fine for provisional but GRPCRoute should probably be supported for implementable. Not blocking right now though IMO

Reply:
I would request to wait until GRPCRoute itself is no longer experimental.   https://gateway-api.sigs.k8s.io/api-types/grpcroute/

 Is that acceptable?

Reply:
I think this may actually be pretty straightforward and identical to HTTP as @howardjohn suggested. @gnossen can correct me here, but I think we should expect the same config from TLS from Gateway -> Backend to apply equally to HTTPRoute and GRPCRoute. 

As far as the experimental status of GRPCRoute, it's hard to get right. We've gotten into situations in upstream Kubernetes where we let in a bunch of incompatible alpha features and it became quite a pain to graduate any of them past alpha. In this case with GRPCRoute being fairly stable at this point, I'd rather cover it here unless it ends up adding significant complexity.

Reply:
I will mark it "implementation-dependent" intstead of "Yes" in the "Connect to backend with TLS?" column.

Reply:
I agree with @robscott . Disregarding any issues of experimental state, I think the values for GRPCRoute should be the same as for HTTPRoute in this table.